### PR TITLE
Add YAML config file per testcase

### PR DESCRIPTION
**Pull request by niemela**
...to allow "full feedback" on test cases.

### CODE REVIEW

**Review by pehrsoderman — APPROVED** [review]

**Comment by pehrsoderman** [comment]
Needs a rebase.

**Comment by RagnarGrootKoerkamp** [comment]
We'd like to move testcase.args, testcase.hint and testcase.description into the testcase.yaml.as well

File: spec/2023-07-draft.md [hunk 1]
```diff
@@ -499,6 +500,26 @@ illustration | image     | `.png`, `.jpg`, `.jpeg`, or `.svg` | privileged infor
...
+Keys are optional unless explicitly stated.
+Any unknown keys should be treated as an error.
+
+Key           | Type    | Default
+------------- | ------- | -------
+full_feedback | Boolean | Inherited from [`testdata.yaml`](#test-data-groups), which defaults to `false` in `secret` and `true` in `sample`
+
+When `full_feedback` is `true` somebody whose submission didn't pass should be shown:
+- the given input,
+- the produced ouput (stdout),
```
Comment (simonlindholm) [inline spec/2023-07-draft.md:518]:
```suggestion
- the produced output (stdout),
```

File: spec/2023-07-draft.md [hunk 2]
```diff
@@ -499,6 +500,26 @@ illustration | image     | `.png`, `.jpg`, `.jpeg`, or `.svg` | privileged infor
...
+- any error messages (stderr),
+- the illustration created by the output visualizer (if applicable),
+- the expected ouput.
```
Comment (simonlindholm) [inline spec/2023-07-draft.md:521]:
```suggestion
- the expected output.
```

**Review by simonlindholm — COMMENTED** [review]
Seems natural to move input and output validator flags to test case level as well (though may not be worth the effort)

**Pull request merged**